sound/soc/sof/sof-priv.h

```diff
@@ -122,11 +122,11 @@ struct snd_sof_dsp_ops {
 		      void __iomem *addr); /* optional */
 
 	/* memcpy IO */
-	void (*block_read)(struct snd_sof_dev *sof_dev, u32 bar,
-			   u32 offset, void *dest,
+	void (*block_read)(struct snd_sof_dev *sof_dev, unsigned int bar,
+			   unsigned int offset, void *dest,
 			   size_t size); /* mandatory */
-	void (*block_write)(struct snd_sof_dev *sof_dev, u32 bar,
-			    u32 offset, void *src,
+	void (*block_write)(struct snd_sof_dev *sof_dev, unsigned int bar,
```

Comment:
I think 'bar' is a strongly typed variable.

Reply:
same - it's an index into `sdev->bar[]`